src/main/java/duck/task/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package duck.task;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import duck.Parser;
+import duck.exception.DuckException;
+
+
+
+
+/**
+ * Class to handle all operations regarding the current list of tasks
+ * by the user.
+ * Implements Serializable to be easily stored in a file.
+ */
+public class TaskList implements Serializable {
+    private List<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();

```

Comment:
**this.** can be removed for better readability

Reply:
Thanks! I'll remove all the uses that don't disambiguate anything.